src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebRequestPSCmdlet.Common.cs

```diff
@@ -1396,6 +1429,109 @@ protected override void StopProcessing()
 
         #region Helper Methods
 
+        
+        /// <summary>
+        /// Adds content to a <see cref="MultipartFormDataContent" />. Object type detection is used to determine if the value is String, File, or Collection.
+        /// </summary>
+        /// <param name="fieldName">The Field Name to use.</param>
+        /// <param name="fieldValue">The Field Value to use.</param>
+        /// <param name="formData">The <see cref="MultipartFormDataContent" />> to update.</param>
+        /// <param name="enumerate">If true, collection types in <paramref name="fieldValue" /> will be enumerated. If false, collections will be treated as single value.</param>
+        internal void AddMultipartContent (object fieldName, object fieldValue, MultipartFormDataContent formData, bool enumerate)
+        {
+            if (null == formData)
+            {
+                formData = new MultipartFormDataContent();
```

Comment:
Is `formData` a return (result) data? Shouldn't it be `out` parameter? I am confused that we use the constructor inside the method and outside.

Reply:
I guess this should throw? The thing is, I didn't want it to throw and out requires assignment. I don't care if someone supplies null to formData but I didn't want the code to fail if they did. maybe I should just return on or an assert?

Reply:
This is only a helper function - I think it is enough to use `Assert()`.

Reply:
fixed. we are throwing `ArgumentNullException` in other methods, I changed it to the same.